admin/app/controllers/solidus_admin/controller_helpers/turbo_actions.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+module SolidusAdmin::ControllerHelpers::TurboActions
+  extend ActiveSupport::Concern
+
+  included do
+    class_attribute :registered_turbo_actions, instance_writer: false, default: []
+    before_action :ensure_turbo_frame_request, if: ->(controller) do
+      registered_turbo_actions.include?(controller.action_name.to_sym)
+    end
+  end
+
+  class_methods do
+    def turbo_actions(*actions)
+      self.registered_turbo_actions += actions.map(&:to_sym)
+    end
+  end
+
+  private
+
+  def ensure_turbo_frame_request
+    redirect_to action: :index unless turbo_frame_request?
+  end
+end
```

Comment:
I think this is not necessary. The ultimate goal is that we still are able to open these routes in ie. a new tab or in tests w/o javascript.

Reply:
Addressed in #6046